crates/compiler/build/src/program.rs

```diff
@@ -156,33 +157,46 @@ fn report_problems_help(
     }
 }
 
+pub enum CodeObject {
+    MemoryBuffer(MemoryBuffer),
```

Comment:
I looked, but it's very unsafe to turn a memory buffer into a `Vec`. We could do something with an existential here, but this seems simpler and really not that inefficient.

Reply:
If they both can be used as slices, that sounds fine.